tests/test_aging.c

```diff
@@ -0,0 +1,532 @@
+ * @brief tests for aging utility
+ *
+ * @copyright
+ * Copyright (c) 2018 - 2022 Deutsche Telekom AG.
+ * Copyright (c) 2018 - 2022 CESNET, z.s.p.o.
+ *
+ * This source code is licensed under BSD 3-Clause License (the "License").
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://opensource.org/licenses/BSD-3-Clause
+ */
+
+#define _GNU_SOURCE
+
+#include <stdio.h>
+#include <string.h>
+#include <dirent.h>
+#include <unistd.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <sys/wait.h>
+#include <signal.h>
+#include <errno.h>
+#include <stdlib.h>
+#include <setjmp.h>
+#include <stdarg.h>
+#include <cmocka.h>
+
+#include <libyang/libyang.h>
+
+#include "sysrepo.h"
+#include "sysrepo_types.h"
+#include "test_common.h"
+#include "config.h"
+
+#define TIME_MAX 100000
+#define NUM_OF_FILES 30
+
+typedef struct {
+    sr_conn_ctx_t *conn;
+    sr_session_ctx_t *sess;
+    char *n_path;
+    char *a_path;
+    char *path_to_file;
+    char *pidfile;
+    uint32_t len_npath;
+    uint32_t len_apath;
+    pid_t pid;
+} test_data_t;
+
+/* Allocs memory, memory needs to be freed */
+static char *
+get_test_path(char *which)
+{
+	char *path;
+	if (asprintf(&path, "%s/data/%s", sr_get_repo_path(), which) == -1) {
+		path = NULL;
+	}
+	return path;
+}
+
+/* from src/common.c */
+static int
+mkpath(const char *path, mode_t mode)
+{
+    char *p, *dup;
+
+    dup = strdup(path);
+    for (p = strchr(dup + 1, '/'); p; p = strchr(p + 1, '/')) {
+        *p = '\0';
+        if (mkdir(dup, mode) == -1) {
+            if (errno != EEXIST) {
+                *p = '/';
+                free(dup);
+                return -1;
+            }
+        }
+        *p = '/';
+    }
+    free(dup);
+
+    if (mkdir(path, mode) == -1) {
+        if (errno != EEXIST) {
+            return -1;
+        }
+    }
+
+    return 0;
+}
+
+static int
+create_file(char *path, char *content)
+{
+    FILE *fp;
+    fp = fopen(path, "w+");
+    chmod(path, 0700);
+    if (!fp) {
+        fprintf(stderr, "Creating a file %s failed\n", path);
+        return EXIT_FAILURE;
+    }
+    if (content) {
+        fprintf(fp, "%s", content);
+    }
+    fclose(fp);
+    return 0;
+}
+
+static int
+remove_file(void **state, char *file_name, int notif_dir)
+{
+    test_data_t *data = (test_data_t *) (*state);
+    if (asprintf(&data->path_to_file, "%s%s", (notif_dir) ? data->n_path : data->a_path, file_name) == -1) {
+        fprintf(stderr, "Asprintf failed\n");
+    }
+    if (remove(data->path_to_file)) {
+        fprintf(stderr, "Removing a file %s failed!\n", file_name);
+    }
+    free(data->path_to_file);
+    data->path_to_file = NULL;
+    return 0;
+}
+
+static int
+create_config(sr_session_ctx_t *sess, char *older_than, char *archive_dir, char *compress)
+{
+    int rc;
+    const struct ly_ctx *ctx;
+    struct lyd_node *node = NULL;
+
+    ctx = sr_session_acquire_context(sess);
```

Comment:
On all errors you need to call `sr_session_release_context(sess);`.

Reply:
FIxed